### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by benjarobin**

### CODE REVIEW

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 1]
```diff
@@ -15,8 +15,7 @@
 #include "socket-util.h"
 #include "strv.h"
 
-#define GET_CONTAINER(m, i) ((i) < (m)->n_containers ? (struct rtattr*)((uint8_t*)(m)->hdr + (m)->containers[i].offset) : NULL)
-#define PUSH_CONTAINER(m, new) (m)->container_offsets[(m)->n_containers++] = (uint8_t*)(new) - (uint8_t*)(m)->hdr;
+#define GET_CONTAINER(m, i) ((struct rtattr*)((uint8_t*)(m)->hdr + (m)->containers[i].offset))
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c:18]: why this change? the old macro looks more robust? can you elaborate?

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 2]
```diff
@@ -520,7 +519,7 @@ int sd_netlink_message_open_container(sd_netlink_message *m, unsigned short type
 
         assert_return(m, -EINVAL);
         assert_return(!m->sealed, -EPERM);
-        assert_return(m->n_containers < RTNL_CONTAINER_DEPTH, -ERANGE);
+        assert_return(m->n_containers < (RTNL_CONTAINER_DEPTH - 1), -ERANGE);
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c:523]: hmm, can you explain?

File: src/systemctl/systemctl.c [hunk 3]
```diff
@@ -3775,6 +3775,8 @@ static int clean_or_freeze_unit(int argc, char *argv[], void *userdata) {
                 method = "FreezeUnit";
         else if (streq(argv[0], "thaw"))
                 method = "ThawUnit";
+        else
+                assert(false);
```
Comment (poettering) [inline src/systemctl/systemctl.c]: assert_not_reached(…)

File: src/journal/test-audit-type.c [hunk 4]
```diff
@@ -20,6 +20,7 @@ static void test_audit_type(void) {
                 print_audit_label(i);
 }
 
-int main(int argc, char **argv) {
+int main(void) {
```
Comment (poettering) [inline src/journal/test-audit-type.c]: uh?

File: src/libsystemd-network/test-sd-dhcp-lease.c [hunk 5]
```diff
@@ -80,11 +80,12 @@ static void test_dhcp_lease_parse_search_domains_wrong_len(void) {
         assert_se(dhcp_lease_parse_search_domains(optionbuf, sizeof(optionbuf) - 5, &domains) == -EBADMSG);
 }
 
-int main(int argc, char *argv[]) {
+int main(void) {
```
Comment (poettering) [inline src/libsystemd-network/test-sd-dhcp-lease.c]: uh?

File: src/libsystemd/sd-netlink/netlink-message.c (hunk 1 above)
Comment (benjarobin) [reply src/libsystemd/sd-netlink/netlink-message.c:18]: - `PUSH_CONTAINER()` is unused, and maybe not in sync with code, too dangerous to keep it.
 - Nobody check the return value of `GET_CONTAINER()` (no check for NULL), and this macro is always called with a previous check on `i` with `n_containers`.

File: src/libsystemd/sd-netlink/netlink-message.c (hunk 2 above)
Comment (benjarobin) [reply src/libsystemd/sd-netlink/netlink-message.c:523]: `m->containers[m->n_containers + 1]` is accessed both in read and write. If `m->n_containers` is equal to `(RTNL_CONTAINER_DEPTH - 1)` the array is accessed out of bound

File: src/systemctl/systemctl.c (hunk 3 above)
Comment (benjarobin) [reply src/systemctl/systemctl.c]: Ok will update it, thanks

File: src/journal/test-audit-type.c (hunk 4 above)
Comment (benjarobin) [reply src/journal/test-audit-type.c]: Some test function are declared like this, this is allowed, and since the arguments are not used... I can revert this part if you want

File: src/libsystemd-network/test-sd-dhcp-lease.c (hunk 5 above)
Comment (benjarobin) [reply src/libsystemd-network/test-sd-dhcp-lease.c]: See https://github.com/systemd/systemd/pull/15765#discussion_r423089306

File: src/journal/test-audit-type.c (hunk 4 above)
Comment (poettering) [reply src/journal/test-audit-type.c]: yes, please leave the normal function prototype in place here. I am aware that the prototype may differ, but it's always a bit ugly, since it just ignores the parameters that are passed anyway. That that works is kind of an ugly C feature if you ask me...

File: src/libsystemd/sd-netlink/netlink-message.c (hunk 2 above)
Comment (poettering) [reply src/libsystemd/sd-netlink/netlink-message.c:523]: please add this as comment here. At least on the first occasion this appears.

File: src/libsystemd/sd-netlink/netlink-message.c (hunk 1 above)
Comment (poettering) [reply src/libsystemd/sd-netlink/netlink-message.c:18]: ok

**Comment by benjarobin** [comment]
@poettering Force pushed, and rebased
